ovn-org/ovn-kubernetes
# Reconcile node lbs on node deletion

**- What this PR does and why is it needed**

services controller: reconcile on node deletion

A node load balancer is a load balancer associated to any of the following
services:
- service with NodePort set
- service with host-network endpoints
- service with ExternalTrafficPolicy=Local
- service with InternalTrafficPolicy=Local

This commit forces the reconciliation of the load balancers upon node
deletion. Since the node was deleted, the newly generated list of load
balancers will *not* feature the deleted node's load balancers, which will
lead to their deletion.

Leaking these load balancers causes an issue when the deleted node is re-added
(as part of a remediation procedure, for instance) since the newly created node
logical switch does not point to these node load balancers, thus breaking
connectivity of pods on the node to the services described in the list above.

closes: https://bugzilla.redhat.com/show_bug.cgi?id=2068910

**- Special notes for reviewers**

Forced service reconciliation when a node is deleted, to ensure the load balancers
associated to the node were deleted.

**- How to verify it**

1. Backup one of your node's configuration - `kubectl get node <nodeXYZ> -oyaml > nodexyz-backup.yaml`
2. Delete said node - `kubectl delete node <nodeXYZ>`
3. Ensure the node load balancers are **deleted** from the OVN NB database. These end with the `nodeXYZ` suffix
4. Add the node back - `kubectl apply -f nodexyz-backup.yaml`
5. Ensure the node load balancers are **found** from the OVN NB database. These end with the `nodeXYZ` suffix

**- Description for the changelog**

Force service reconciliation whenever a node is deleted.
